src/jit/lower.cpp

```diff
@@ -3198,7 +3196,9 @@ GenTree* Lowering::LowerVirtualStubCall(GenTreeCall* call)
 #ifndef _TARGET_X86_
             // on x64 we must materialize the target using specific registers.
             addr->gtRegNum  = REG_VIRTUAL_STUB_PARAM;
+
             indir->gtRegNum = REG_JUMP_THUNK_PARAM;
+            indir->gtFlags |= GTF_IND_VSD_TGT;
 #endif
```

Comment:
Note that non-standard arg to move stub cell addr in R11 gets added to call node only for indirect VSD calls on amd64.
In case of direct VSD calls on amd64, it depends on setting TreeNodeInfo::Initialize() is setting DstCandidates on addr node.
TreeNodeInfo::Initialize() in turn sets DstCandidates only if gtRegNum field is not REG_NA.

For direct VSD calls on amd64, Like in x86 case, we should be setting srcandidates on addr node in lowerxarch.cpp and in codegenxarch.cpp, should be doing a reg move if addr is not already in the right register.

I would leave it up to you whether you want to fix that with this PR or file a tracking issue for it.